src/lib/site-server-process.ts

```diff
@@ -142,14 +143,26 @@ export default class SiteServerProcess {
 		}
 
 		return new Promise< void >( ( resolve, reject ) => {
+			let isResolved = false;
 			process.once( 'exit', ( code ) => {
-				if ( code !== 0 ) {
-					reject( new Error( `Site server process exited with code ${ code } upon stopping` ) );
-					return;
+				if ( ! isResolved ) {
+					isResolved = true;
+					if ( code !== 0 ) {
+						reject( new Error( `Site server process exited with code ${ code } upon stopping` ) );
+					} else {
+						resolve();
+					}
 				}
-				resolve();
 			} );
-			process.kill();
+			// process.kill() returns false if we're not able to kill the process.
+			if ( ! process.kill() && ! isResolved ) {
```

Comment:
```suggestion
			if ( ! process.kill() ) {
```
I can't really imagine a scenario in which `isResolved` would be true at this point 🤔

Reply:
I guess I was being extra careful with that condition, but it doesn't make sense anymore with the new approach.
I have removed `isResolved` check and works as expected.
Thanks for the feedback!